oauth2_http/java/com/google/auth/oauth2/ExternalAccountCredentials.java

```diff
@@ -97,7 +99,7 @@ abstract static class CredentialSource {
 
   protected transient HttpTransportFactory transportFactory;
 
-  @Nullable protected final ImpersonatedCredentials impersonatedCredentials;
+  @Nullable protected ImpersonatedCredentials impersonatedCredentials;
```

Comment:
Since it is being called in the base class constructor, the underlying credentials are being initialized before the handler is being set. We need to call the method again after setting the handler in the subclass.